### ISSUE TITLE
How to use "check point" events?

### ISSUE THREAD

**Issue by iSunRise** [storeon/undo#18]
Hello,

Thank you for amazing library.

I'm wondering, is there a way to add new history record only after specific event is dispatched?

For example, I'm dragging some element and update its position on each mouse move, however, I don't want all those position to appear in history. Instead I want to commit only after "drag end", so I have only final positions in history.

 P.S. After typing this message, I reviewed the source code and found out that history records are created on "@changed" event unless the updates are filtered.

Would you consider to extend the library?

Thanks

**Comment by polemius** [comment]
@iSunRise 

Hello 👋 

I'm happy that you like this library.

If for example you will have the parameters in function `createHistory` to exclude some events, it can help you?

> I'm dragging some element and update its position on each mouse move
> Instead I want to commit only after "drag end", so I have only final positions in history.

What if you will store each mouse move position in localstate of component and only store final position when drag ends? It will be possible to implement?

**Comment by iSunRise** [comment]
hi @polemius,

> What if you will store each mouse move position in localstate of component and only store final position when drag ends? It will be possible to implement?

Yes, it is probably the easiest and performant solution, but it works only if other components don't have to be aware of my draggable component temporary position. 

> If for example you will have the parameters in function createHistory to exclude some events, it can help you?

I think that "filter events/exclude events" params would bring much more flexibility, however, I've made my own solution to handle my needs already.

Thank you for the quick response, the issue may be closed now.

**Comment by polemius** [comment]
Hello @iSunRise,

I see. Thank you for response. I've create separate issue for this #16 if you want fell free to open PR.
